### DISCUSSION TITLE
Compatibility with Next.js 13 Route Handlers

### DISCUSSION THREAD

**Discussion by sarimabbas** [ecyrbe/zodios#424] [Q&A]
Hello! I hope you are well. Thanks so much for your work on this library. 

It seems like [Next.js Route Handlers](https://beta.nextjs.org/docs/routing/route-handlers) run in both Node.js and Edge runtimes, with isomorphic Request and Response objects. Is it possible to set up the server component of Zodios with a Next.js route handler instead of a Next.js API route? It seems like a Fetch-based client implementation is landing in v11. Will something like that follow on the server? Perhaps it is too difficult if the Zodios app uses Express under the hood. Would love to hear your thoughts, thank you!

**Comment by ecyrbe** [comment]
Unfortunately, route handlers would not allow zodios to infer the route, because ts don't have access to file named information.
Converting to conversation for Q&A

**Comment by ecyrbe** [comment]
You can still implement your server with next route handlers and use zodios for client side only (or call external api on server side) . Just that you'll not get server side type safety

**Reply by sarimabbas** [reply]
Thanks for your response and that sounds good! Currently, using something else to set up OpenAPI server, and using Zodios for the client is sufficient for me
